darts/models/regression_model.py

```diff
@@ -13,300 +13,259 @@
+        lags_exog : Union[int, list, bool]
+            Number of lagged exogenous values used to predict the next time step. If an integer is given
+            the last `lags_exog` lags are used (inclusive). Otherwise a list of integers with lags. If False,
+            the value at time `t` is used which might lead to leakage. If True the same lags as for the
+            target variable are used.
+        model
+            A regression model that implements `fit()` and `predict()` methods.
+            Default: `sklearn.linear_model.LinearRegression(n_jobs=-1, fit_intercept=False)`
+        """
+        raise_if((lags is None) and (lags_exog is None),
+            "At least one of 'lags' or 'lags_exog' must be not None."
+        )
+        raise_if_not(isinstance(lags, (int, list)) or lags is None,
+            "`lags` must be of type int or list."
+        )
+        raise_if_not(isinstance(lags_exog, (int, list, bool)) or lags_exog is None,
+            "`lags_exog` must be of type int, list or bool."
+        )
+        raise_if(lags is None and lags_exog is True,
+            "`lags_exog` must not be True if `lags` is None."
+        )
+
+        if (not callable(getattr(model, "fit", None))):
+            raise_log(Exception('Provided model object must have a fit() method', logger))
+        if (not callable(getattr(model, "predict", None))):
+            raise_log(Exception('Provided model object must have a predict() method', logger))
+
+        if model is None:
+            model = LinearRegression(n_jobs=-1, fit_intercept=False)
+
+        self.lags = lags
+        if isinstance(self.lags, int):
+            raise_if_not(self.lags > 0, "`lags` must be strictly positive. Given: {}.".format(self.lags))
+            self.lags = list(range(1, self.lags+1))
+        elif isinstance(self.lags, list):
+            for lag in self.lags:
+                raise_if(not isinstance(lag, int) or (lag <= 0), (
+                    "Every element of `lags` must be a strictly positive integer. Given: {}.".format(self.lags))
+                )
+
+        self.lags_exog = lags_exog
+        if self.lags_exog is True:
+            self.lags_exog = self.lags[:]
+        elif self.lags_exog is False:
+            self.lags_exog = [0]
+        elif isinstance(self.lags_exog, int):
+            raise_if_not(self.lags_exog > 0, "`lags_exog` must be strictly positive. Given: {}.".format(self.lags_exog))
+            self.lags_exog = list(range(1, self.lags_exog+1))
+        elif isinstance(self.lags_exog, list):
+            for lag in self.lags_exog:
+                raise_if(not isinstance(lag, int) or (lag < 0), (
+                    "Every element of `lags_exog` must be a positive integer. Given: {}."
+                    .format(self.lags_exog))
+                )
+
+        self.model = model
+        if self.lags is not None and self.lags_exog is None:
+            self.max_lag = int(np.max(self.lags))
```

Comment:
you can also simply use the built-in `max()` instead of `np.max()`.

Reply:
Done